### PR TITLE
Leaflet map

### PR DESCRIPTION
**Pull request by Natkuma01**
# Description
Implemented a drop-down menu that allows users to select restaurants by state. When the user clicks the "Search" button, they are directed to another page that displays the restaurants with markers on a map. The markers are automatically centered in the middle of the map. When users hover over the markers, the name of the restaurant is displayed. On top of the map, it has categories like Steak, Burger, Dessert, Pizza, and Fried Chicken. When the user clicks on the image, it filters out the restaurants located in that specific state and belongs to the selected category.

## Main changes explained:

- created Map directory included FilterByCategory.js, FilterState.js, and RestaurantByState.js
- added drop-down menu for State selection
- added category images to filter restaurants by category
- installed react-router, leaflet-react, leaflet, and axios
- created a map to display the locations of restaurants.

## How to test:
1. check into current branch
2. do `npm install` and `npm start` to run this PR locally
3. go to localhost:3000
4. select a state from drop-down menu and click 'Search' button
5. hover to any restaurants on the map and click on it

## Screenshots or videos of changes:
Screenshot 2024-08-08 at 1 15 10 AM

Screenshot 2024-08-08 at 1 12 35 AM

## Note:
Need to improve the interface - The color for the drop-down menu looks different from Figma.

### CODE REVIEW

File: src/components/maps.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                    Array.isArray(item) ? item : [item])
+
+                // push data we needed to use in allRestaurants Array
+                const allRestaurants = []
+                normalized.forEach(category => {
+                    category.forEach(restaurant => {
+                      if (restaurant.country) {
+                        const restaurantState = restaurant.country.split(', ').pop()
+                        // no unique id in the data
+                        const id = restaurant.index
```
Comment (cvtqx) [inline src/components/maps.js]: Suggestion - use Math.random() to generate a unique id or uuid library. with index id comes as undefined when I log the restaurants list:

Screenshot 2024-08-07 at 10 03 24 PM

File: src/components/maps.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import steak from "../images/category/steak.png"
+import desserts from "../images/category/desserts.png"
+import { Icon } from "leaflet"
+import axios from 'axios'
+import 'leaflet/dist/leaflet.css'
+
+const MapPage = () => {
+    const { state } = useParams()
+    const [restaurants, setRestaurants] = useState([])
+    const navigate = useNavigate
```
Comment (cvtqx) [inline src/components/maps.js]: This should be `const navigate = useNavigate()`. but I am not sure how it is used in this code?

**Review by cvtqx — COMMENTED** [review]
Great job, Natalie! the map filters by state and displays restaurants on the map. I have added a couple of minor suggestions on some lines of code, please check. 

Other suggestions:

- put maps and filterstate components into a folder with a good name so everyone knows where they belong(but I see your comment on top of maps.js, which perhaps means you were going to add it into Pages?). perhaps Pages or MainPage...

- add styling to the input field and button. see if you find anything in Daisy UI for the input field. and add theme colors. you can find theme colors in tailwind.config.

File: src/components/maps.js (hunk 2 above)
Comment (Natkuma01) [reply src/components/maps.js]: That's a good catch! I was wondering why it cannot direct me to other page.

File: src/components/maps.js (hunk 1 above)
Comment (Natkuma01) [reply src/components/maps.js]: Thank you. This is such a great idea! I will go ahead do that

**Review by cvtqx — APPROVED** [review]
Looks great, Natalie! Let's merge it!

**Pull request merged**